### ISSUE TITLE
Error and no output because of Notice: Array to string conversion

### ISSUE THREAD

**Issue by Tantuss** [magmodules/magento2-channable#8]
a:4:{i:0;s:124:"Notice: Array to string conversion in /vendor/magmodules/magento2-channable/Helper/Product.php on line 375";i:1;s:10584:"#0 /vendor/magmodules/magento2-channable/Helper/Product.php(375): Magento\Framework\App\ErrorHandler->handler(8, 'Array to string...', '/data/web/magen...', 375, Array)
tice: Array to string conversion in /vendor/magmodules/magento2-channable/Helper/Product.php on line 375

Issue is probably because of selecting 'meta_robots' as field.
This is not logical, but anyway, it should not create an error.

Issue in code:
```
        if ($attribute['type'] == 'select') {
            if ($attr = $product->getResource()->getAttribute($attribute['source'])) {
                $value = $product->getData($attribute['source']);
                return (string)$attr->getSource()->getOptionText($value);
            }
        }
```

Proposal to fix:
```
        if ($attribute['type'] == 'select') {
            if ($attr = $product->getResource()->getAttribute($attribute['source'])) {
                $value = $product->getData($attribute['source']);
                $data = $attr->getSource()->getOptionText($value);
                if(is_array($data)){
                    if(count($data) > 0) return current($data);
                    return '';
                }
                return (string)$data;
            }
        }
```

**Comment by magmodules** [comment]
Hi! Attribute with type select should not return an array, this can only happen when using type multiselect. Can meta_robots have multiple values? Was this attribute added by an extension or did you create this manually thought the Magento Admin?

**Comment by Tantuss** [comment]
Good question, the Magento which I started using was already setup with some extension.
And it was a migration from 1.9 to 2.1.

So it might almost never happen, but apparently it is possible, that's why I added the suggested fix.

**Comment by Tantuss** [comment]
It doesn't have any options to select.
But the strange thing is that the values are written into the table: catalog_product_entity_text

So looks like this is also a migration issue.

**Comment by magmodules** [comment]
Fixed in 1.0.2
